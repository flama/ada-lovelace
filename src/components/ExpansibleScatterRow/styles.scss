@import "../base";

.expansible-scatter-row {
  overflow: hidden;
  display: flex;
  flex-direction: column;

  height: 100%;
  transition: all 0.5s ease-in-out;

  &.hidden {
    overflow: hidden;
    height: 0;
  }
}
