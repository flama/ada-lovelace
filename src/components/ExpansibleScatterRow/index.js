import React, { Component } from 'react'
import ScatterRow from '../ScatterRow'

import './styles.scss'

class ExpansibleScatterRow extends Component {
  classNames = () => {
    return `expansible-scatter-row ${!this.props.data.active && !this.props.all ? 'hidden':''}`
  }

  render() {
    let topics, subtopics

    if(this.props.data.active) {
      subtopics = Object.keys(this.props.data.division).map(subcategory => {
        return (<ScatterRow
          data={ this.props.data.division[subcategory] || [] }
          domain={ this.props.domain }
          title={ subcategory }
          hidden={ true }
          key={ subcategory }
          />)
        })
    } else {
      topics = (<ScatterRow
        data={ this.props.data.array }
        domain={ this.props.domain }
        title={ this.props.title }
        hidden={ !this.props.all }
      />)
    }


    return (
      <div className={ this.classNames() }>
        { topics || subtopics }
      </div>
    )
  }
}

ExpansibleScatterRow.propTypes = {
  data: React.PropTypes.object,
  domain: React.PropTypes.object,
  title: React.PropTypes.string,
  hidden: React.PropTypes.bool,
  all: React.PropTypes.bool
}

export default ExpansibleScatterRow
