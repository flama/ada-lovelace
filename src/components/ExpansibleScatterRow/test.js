import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import ExpansibleScatterRow from '../ExpansibleScatterRow'

// TODO: test functionality

describe("ExpansibleScatterRow", () => {
  it("should render categories corretly", () => {
    let tree = shallow(
      <ExpansibleScatterRow
        data={{
          active: false,
          division: {},
          array: []
        }}
        domain={{ x: [1800, 2020] }}
      />
    )
    expect(tree).toMatchSnapshot()
  })

  // TODO: move shallow to beforeEach, use setProps to change active prop
  it("should render subcategories correctly", () => {
    let tree = shallow(
      <ExpansibleScatterRow
        data={{
          active: true,
          division: {},
          array: []
        }}
        domain={{ x: [1800, 2020] }}
      />
    )
    expect(tree).toMatchSnapshot()
  })
})
