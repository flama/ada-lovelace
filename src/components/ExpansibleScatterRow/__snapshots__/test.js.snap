// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpansibleScatterRow should render categories corretly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="expansible-scatter-row hidden"
>
     
    <ScatterRow
        data={Array []}
        domain={
            Object {
                "x": Array [
                  1800,
                  2020,
                ],
              }
        }
        hidden={true}
        title={undefined}
    />
     
</div>,
  "nodes": Array [
    <div
      className="expansible-scatter-row hidden"
>
       
      <ScatterRow
            data={Array []}
            domain={
                  Object {
                        "x": Array [
                          1800,
                          2020,
                        ],
                      }
            }
            hidden={true}
            title={undefined}
      />
       
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ExpansibleScatterRow
        data={
                Object {
                        "active": false,
                        "array": Array [],
                        "division": Object {},
                      }
        }
        domain={
                Object {
                        "x": Array [
                          1800,
                          2020,
                        ],
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "data": Object {
            "active": false,
            "array": Array [],
            "division": Object {},
          },
          "domain": Object {
            "x": Array [
              1800,
              2020,
            ],
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="expansible-scatter-row hidden"
>
           
          <ScatterRow
                    data={Array []}
                    domain={
                              Object {
                                        "x": Array [
                                          1800,
                                          2020,
                                        ],
                                      }
                    }
                    hidden={true}
                    title={undefined}
          />
           
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="expansible-scatter-row hidden"
>
           
          <ScatterRow
                    data={Array []}
                    domain={
                              Object {
                                        "x": Array [
                                          1800,
                                          2020,
                                        ],
                                      }
                    }
                    hidden={true}
                    title={undefined}
          />
           
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ExpansibleScatterRow
    data={
        Object {
            "active": false,
            "array": Array [],
            "division": Object {},
          }
    }
    domain={
        Object {
            "x": Array [
              1800,
              2020,
            ],
          }
    }
/>,
}
`;

exports[`ExpansibleScatterRow should render subcategories correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="expansible-scatter-row "
>
     
     
</div>,
  "nodes": Array [
    <div
      className="expansible-scatter-row "
>
       
       
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ExpansibleScatterRow
        data={
                Object {
                        "active": true,
                        "array": Array [],
                        "division": Object {},
                      }
        }
        domain={
                Object {
                        "x": Array [
                          1800,
                          2020,
                        ],
                      }
        }
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "data": Object {
            "active": true,
            "array": Array [],
            "division": Object {},
          },
          "domain": Object {
            "x": Array [
              1800,
              2020,
            ],
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="expansible-scatter-row "
>
           
           
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="expansible-scatter-row "
>
           
           
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ExpansibleScatterRow
    data={
        Object {
            "active": true,
            "array": Array [],
            "division": Object {},
          }
    }
    domain={
        Object {
            "x": Array [
              1800,
              2020,
            ],
          }
    }
/>,
}
`;
